File: solution/0300-0399/0376.Wiggle Subsequence/Solution.py
class Solution:
    def wiggleMaxLength(self, nums: List[int]) -> int:
        f = g = 1
        for a, b in pairwise(nums):
            if a < b:
                f = max(f, g + 1)
            if a > b:
                g = max(g, f + 1)
        return max(f, g)

File: solution/0300-0399/0376.Wiggle Subsequence/Solution.cpp
class Solution {
public:
    int wiggleMaxLength(vector<int>& nums) {
        int f = 1, g = 1;
        for (int i = 1; i < nums.size(); ++i) {
            int d = nums[i] - nums[i - 1];
            if (d < 0) {
                f = max(f, g + 1);
            }
            if (d > 0) {
                g = max(g, f + 1);
            }
        }
        return max(f, g);
    }
};